lib/Dialect/FIRRTL/Ops.cpp

```diff
@@ -642,9 +642,14 @@ MemOp::getTypeForPortList(uint64_t depth, FIRRTLType dataType,
 /// Return the kind of port this is given the port type from a 'mem' decl.
 static Optional<MemOp::PortKind> getMemPortKindFromType(FIRRTLType type) {
   auto portType = type.dyn_cast<BundleType>();
+  if (!portType) {
+    auto flipType = type.dyn_cast<FlipType>();
```

Comment:
Super nit, but you can fold the flipType decl into the if, like `    if (auto flipType = type.dyn_cast<FlipType>())`

either way is fine though.